src/cpp/ext/otel/otel_plugin.cc

```diff
@@ -150,15 +150,19 @@ OpenTelemetryPluginBuilderImpl::SetMeterProvider(
   return *this;
 }
 
-OpenTelemetryPluginBuilderImpl& OpenTelemetryPluginBuilderImpl::EnableMetric(
-    absl::string_view metric_name) {
-  metrics_.emplace(metric_name);
+OpenTelemetryPluginBuilderImpl& OpenTelemetryPluginBuilderImpl::EnableMetrics(
```

Comment:
Not directly related to this PR, but what's stopping us from eliminating `OpenTelemetryPluginBuilderImpl` at this point?

Reply:
Currently, `OpenTelemetryPluginBuilderImpl` has the following additional methods - 
1) `SetTargetSelector` - This will go away after my next PR to add channel scope filter.
2) `SetServerSelector` - This is unused at the moment and can be removed.
3) `TestOnlyEnabledMetrics` - Used for testing to see which metrics are enabled.

Reply:
so, yeah, really it can be removed

Reply:
Okay, please remove it as a follow-up PR.

Reply:
Isn't there some benefits with the Pimpl idiom used here? e.g. if we change the implementation details of the `OpenTelemetryPluginBuilderImpl`, the users of the `OpenTelemetryPluginBuilder` API do not need to recompile their code (as long as the API does not change).

https://en.cppreference.com/w/cpp/language/pimpl